framework/src/wifi_manager/slsi_wifi_utils.c

```diff
@@ -354,6 +407,27 @@ wifi_utils_result_e wifi_utils_connect_ap(wifi_utils_ap_config_s *ap_connect_con
 	slsi_security_config_t *config = NULL;
 
 	if (ap_connect_config->passphrase_length > 0) {
+		// scan to get the security config if it is unknown
```

Comment:
One question from my side;
do we really need (or manage) the additional scanning in this layer?
I guess the slsi WiFi driver might have a process that finds the auth and crypto within the driver layer.
Please verify that feature exists in the driver code or not.

Reply:
I checked slsi WiFi driver, there's no such feature to scan specific ssid. 
I also checked @glb1013's modification in driver, it cannot make any optimization, so I removed his change.